### PR TITLE
Update TiDB version to v5.0

### LINKED ISSUE #3900: Update TiDB version to v5.0
## Feature Request

**Is your feature request related to a problem? Please describe:**

Update the version of the TiDB core components and the tools in the tidb-backup-manager to v5.0. And also the versions in the docs-tidb-operator repo.
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve?

Closes #3900
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Update TiDB version to v5.0
```

### CODE REVIEW

**Review by july2993 — APPROVED** [review]

File: examples/advanced/tidb-cluster.yaml [hunk 1]
```diff
@@ -446,7 +446,7 @@ spec:
     ##   - `command`, which uses curl to access the /status path on port 10080 of the pod
     ## This is supported from TiDB Operator v1.1.7
     # readinessProbe:
-    #   # The `command` type is only supported after tidb v4.0.10, ref: https://github.com/pingcap/tidb/pull/20694
```
Comment (DanielZhangQD) [inline examples/advanced/tidb-cluster.yaml:449]: This should be v4.0.9.

Comment (handlerww) [reply examples/advanced/tidb-cluster.yaml:449]: fixed

File: examples/namespaced/tidb-cluster.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ kind: TidbCluster
 metadata:
   name: basic
 spec:
-  version: v4.0.10
+  version: v5.0.0
```
Comment (DanielZhangQD) [inline examples/namespaced/tidb-cluster.yaml:9]: Please also cover `examples/namespaced/tidb-monitor.yaml`.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: images/tidb-backup-manager/Dockerfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 FROM pingcap/tidb-enterprise-tools:latest
 ARG RCLONE_VERSION=v1.51.0
 ARG SHUSH_VERSION=v1.4.0
-ARG TOOLKIT_V40=v4.0.10
+ARG TOOLKIT_V40=v5.0.0
```
Comment (csuzhangxc) [inline images/tidb-backup-manager/Dockerfile:4]: 😂 will you rename `V40` to `V50`?

**Comment by csuzhangxc** [comment]
maybe we need another E2E test case to upgrade from v4.0 to v5.0?

File: images/tidb-backup-manager/Dockerfile (hunk 3 above)
Comment (DanielZhangQD) [reply images/tidb-backup-manager/Dockerfile:4]: Just rename it to `TOOLKIT_VERSION`?

**Comment by DanielZhangQD** [comment]
> maybe we need another E2E test case to upgrade from v4.0 to v5.0?

@handlerww @csuzhangxc What about cover the case in this issue https://github.com/pingcap/tidb-operator/issues/3877?

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**